frontend/src/pages/PipelineDetails.tsx

```diff
@@ -293,8 +297,33 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
 
     this.props.updateToolbar({ breadcrumbs, actions: toolbarActions, pageTitle });
 
-    if (isFeatureEnabled(FeatureKey.V2) && isV2PipelineSpec(templateString)) {
-      const graphV2 = 'TO BE FULFILLED, non-empty string will open V2';
+    let graph: graphlib.Graph | null = null;
+    let reducedGraph: graphlib.Graph | null | undefined = null;
+    let graphV2: PipelineFlowElement[] = [];
+    if (templateString) {
+      try {
+        const template = JsYaml.safeLoad(templateString);
+        graph = StaticGraphParser.createGraph(template!);
```

Comment:
I'd recommend not relying on create graph throwing an error, because there can be:
1. a v1 pipeline template, but with an error
2. a v2 pipeline template
3. unknown pipeline template

For v1 pipeline template, we can accurately know it's type, by
```
const template = JsYaml.safeLoad(templateString)
if template?.kind != "Workflow" || template?.apiVersion?.startsWith("argoproj.io/") {
  return false
}
return true
```
this gives us a clear idea which templates are v1. for now, we don't need to distinguish v2 templates, the rest are v2 templates.
```

Reply:
Thanks for the detailed logic discussion! I applied the suggestion with some adjustment, it is a good point for using common field in Argo workflow to distinguish v1 template from other templates!